src/orders/components/OrderManualTransactionRefundPage/components/OrderManualTransactionRefundForm/OrderManualTransactionRefundForm.tsx

```diff
@@ -18,27 +19,42 @@ import { useIntl } from "react-intl";
 
 import { getValidationSchema, ManualRefundForm } from "./manualRefundValidationSchema";
 
-interface OrderManualTransationRefundFormProps {
+interface OrderManualTransactionRefundFormProps {
   children: ReactNode;
   disabled: boolean;
   initialValues: ManualRefundForm;
   orderId: string;
+  transactions: TransactionItemFragment[];
 }
 
-export const OrderManualTransationRefundForm = ({
+export const OrderManualTransactionRefundForm = ({
   disabled,
   orderId,
   initialValues,
+  transactions,
   children,
-}: OrderManualTransationRefundFormProps) => {
+}: OrderManualTransactionRefundFormProps) => {
   const intl = useIntl();
   const navigate = useNavigator();
   const notify = useNotifier();
+  const [chargedAmountForValidation, setChargedAmountForValidation] = React.useState(0);
   const methods = useForm<ManualRefundForm>({
     mode: "onBlur",
     values: initialValues,
-    resolver: zodResolver(getValidationSchema(intl)),
+    resolver: zodResolver(getValidationSchema(intl, chargedAmountForValidation)),
   });
+
+  const selectedTransactionId = methods.watch("transationId");
+  const selectedTransaction = transactions.find(
+    transaction => transaction.id === selectedTransactionId,
+  );
+
+  const chargedAmount = selectedTransaction?.chargedAmount?.amount || 0;
+
+  React.useEffect(() => {
+    setChargedAmountForValidation(chargedAmount);
+  }, [chargedAmount]);
```

Comment:
suggestion: there is no reason too keep this in state as it's computed value

Reply:
Yes, but it is computed based on `watch(transactionId)`, which comes from `useForm`. And `useForm` needs this value because there is no method to dynamically set the resolver.

Reply:
+1 this is a computed value why do you need useState for it? why not just use chargeAmount?

Reply:
I think you can solve this by doing refine on the entire schema as you actually have dependent field: amount depends on transactionId